pkg/nas/internal/config.go

```diff
@@ -0,0 +1,163 @@
+		NasClientFactory: cloud.NewNasClientFactory(),
+	}
+
+	cm, err := kubeClient.CoreV1().ConfigMaps(configMapNamespace).Get(context.Background(), configMapName, metav1.GetOptions{})
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			return nil, err
+		}
+	} else {
+		config.SkipSubpathCreation, _ = parseBool(cm.Data["nas-fake-provision"])
+	}
+
+	config.EnableSubpathFinalizer, _ = parseBool(os.Getenv("ENABLE_NAS_SUBPATH_FINALIZER"))
+
+	return config, nil
+}
+
+type NodeConfig struct {
+	NodeName          string
+	NodeIP            string
+	EnableEFCCache    bool
+	EnableMixRuntime  bool
+	EnablePortCheck   bool
+	EnableLosetup     bool
+	EnableVolumeStats bool
+
+	// clients for kubernetes
+	KubeClient kubernetes.Interface
+	CNFSGetter cnfsv1beta1.CNFSGetter
+}
+
+func GetNodeConfig() (*NodeConfig, error) {
+	kubeClient, cnfsGetter := mustGetKubeClients()
+	config := &NodeConfig{
+		// enable nfs port check by default
+		EnablePortCheck: true,
+		KubeClient:      kubeClient,
+		CNFSGetter:      cnfsGetter,
+	}
+
+	// check if enable nfs port check
+	if value := os.Getenv("NAS_PORT_CHECK"); value != "" {
+		config.EnablePortCheck, _ = parseBool(value)
+	}
+
+	// get csi-plugin configmap
+	cm, err := kubeClient.CoreV1().ConfigMaps(configMapNamespace).Get(context.Background(), configMapName, metav1.GetOptions{})
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			return nil, err
+		}
+	} else {
+		if value := cm.Data["nas-metric-enable"]; value != "" {
+			config.EnableVolumeStats, _ = parseBool(value)
+		}
+		config.EnableEFCCache = cm.Data["cnfs-cache-properties"] != "" || cm.Data["nas-efc-cache"] != ""
+	}
+	if value := os.Getenv("NAS_METRIC_BY_PLUGIN"); value != "" {
+		config.EnableVolumeStats, _ = parseBool(value)
+	}
+	if config.EnableVolumeStats {
+		logrus.Info("enabled nas volume stats")
+	}
+
+	// get node info
+	nodeName := os.Getenv("KUBE_NODE_NAME")
+	node, err := kubeClient.CoreV1().Nodes().Get(context.Background(), nodeName, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	config.NodeName = nodeName
+
+	// check container runtime
+	config.EnableMixRuntime = strings.TrimSpace(node.Labels["alibabacloud.com/container-runtime"]) == "Sandboxed-Container.runv" &&
+		strings.HasPrefix(strings.TrimSpace(node.Labels["alibabacloud.com/container-runtime-version"]), "1.")
+	if config.EnableMixRuntime {
+		logrus.Info("enabled nas mix runtime mode")
+	}
+
+	// check if losetup enabled
+	if value := os.Getenv("NAS_LOSETUP_ENABLE"); value != "" {
+		config.EnableLosetup, _ = parseBool(value)
+	}
+	if config.EnableLosetup {
+		logrus.Info("enabled nas losetup mode")
+		for _, addr := range node.Status.Addresses {
+			if addr.Type == corev1.NodeInternalIP {
+				config.NodeIP = addr.Address
+				break
+			}
+		}
+		if config.NodeIP == "" {
+			return nil, errors.New("enabled losetup mode but failed to get node IP")
+		}
+	}
+
+	return config, nil
+}
+
+func parseBool(str string) (bool, error) {
```

Comment:
Can we use `strconv.ParsePool()` instead?

Reply:
No because it can not recognize "enabled" or "yes" which are legal values in our doc.